modules/internal/BytesStringCommon.chpl

```diff
@@ -1243,6 +1244,115 @@ module BytesStringCommon {
     return localX[start..end];
   }
 
+  iter theseBytes(const ref x: ?t) {
+    if x.isEmpty() then return;
+    const localX = x.localize();
+    foreach i in localX.byteIndices do
+      yield localX.byte(i);
+  }
+
+  iter theseBytes(param tag: iterKind, const ref x: ?t)
+      where tag==iterKind.leader {
+
+    const localX = x.localize();
+    for rangeChunk in localX.byteIndices.these(tag) do
+      yield rangeChunk;
+  }
+
+  iter theseBytes(param tag: iterKind, const ref x: ?t, followThis)
+      where tag==iterKind.follower {
+
+    const localX = x.localize();
+
+    for i in localX.byteIndices.these(tag, followThis) do
+      yield localX.buff[i];
+  }
+
+  iter theseAscii(const ref x: ?t) {
+    if x.isEmpty() then return;
+    const localX = x.localize();
+    foreach i in localX.byteIndices {
+      var (newBuff, allocSize) = bufferCopyLocal(localX.buff+i, len=1);
+      if localX.type == string {
+        yield chpl_createStringWithOwnedBufferNV(newBuff, 1, allocSize, 1);
+      }
+      else {
+        yield createBytesWithOwnedBuffer(newBuff, 1, allocSize);
+      }
+    }
+  }
```

Comment:
Am I correct that we could create a standalone version of these iterators by just cloning the local version and changing `foreach` to `forall`?  If so, it seems like it could be worthwhile to remove all the book-keeping involved in leader/follower iterators for a case like `forall codePoints in myString` or `forall byte in myBytes`?  (though also not strictly necessary... mostly what I'm trying to avoid is a situation like the one with `Block` where we've gone years without a standalone Block iterator and I've had a PR open for a crazy long time to add it...  Though the barriers to adding one here later will be lower).